### DISCUSSION TITLE
Headless rendering

### DISCUSSION THREAD

**Discussion by seawerst** [GrapesJS/grapesjs#5454] [Q&A]
In the roadmap discussion #5291, it's mentioned that "The core should be as headless as possible".

I was looking for a way to rerender/reparse the json and the HTML based on an updated collection of components. So a [PATCH version](https://semver.org/spec/v2.0.0.html) of one of our components could refresh saved pages from our CMS via an update background process. I try using nodejs but there are multiple dependencies on browser's features.

Is this somewhere in the roadmap or maybe already possible somehow?
We manage to make something work by automating a chromium webbrowser control inside a .Net console app which is really not ideal.

**Comment by artf** [comment]
>  I try using nodejs but there are multiple dependencies on browser's features.

Can you please indicate the related dependencies?
